course/services/question.py

```diff
@@ -34,3 +34,19 @@ def get_unsolved_practice_questions_count_by_category(user_id):
             result.append({"category": category.id, "difficulty": difficulty, "unsolved_questions": unsolved_questions})
 
     return result
+
+
+def get_number_of_questions_counted_by_category_and_difficulty():
+    categories = QuestionCategory.objects.all()
+
+    result = []
+
+    for category in categories:
+        for difficulty, _ in DIFFICULTY_CHOICES:
+            available_questions = category.question_set.filter(category=category, difficulty=difficulty).count()
```

Comment:
You need to add some more filters. `is_verified=True` , `course=None`, `event=None`, and `question_status=Question.CREATED`.